### DISCUSSION TITLE
Reading and editing pickle files

### DISCUSSION THREAD

**Discussion by Joseph11024** [kwsch/PKHeX#4250] [General]
I've noticed that a majority of the events from 2005-2006 are not in the database. I was trying to make a personal database of all events for every game and cand seem to find most of these and the ones I do find show up illegal. Any help on this would be greatly appreciated.

**Comment by kwsch** [comment]
Nobody has created a serialized listing of Gen3 events that contains all the data necessary to validate a result entity.

Gen4+ has the archive of mystery gift received data collected by the community, and Gen1/2 was manually curated for the few dozen events that were made available. No such table yet exists besides scattered wiki pages with incomplete information.

**Comment by Joseph11024** [comment]
I would like to get some information on editing pickle files. I’m trying to learn new things so I can better help and contribute to the Pokémon community. 
I was hoping to get a better understanding on how this works and what programs, tools and information is needed.

**Reply by kwsch** [reply]
pkl files aren't intended to be edited, only created and consumed. If you open one in a hex editor, like S/V raids or swarms, you'll see that it's essentially a spreadsheet with values. The programs that consume the pkl files read them into objects for logic.

pkl is an arbitrary extension, meaning "PKHeX legality", and is coincidentally similarly meaning & purpose to python's pickle where it converts objects to data and vice versa.

Basically we give the program a database, and an easy way with little overhead is to just read and write objects. The format just has to match for read & write, otherwise you end up with misaligned / incorrect data. Pretty sure there's a previous discussion / issue asking about pickles.

**Reply by Joseph11024** [reply]
When I say edit I mean like updating for new Pokémon for example a new raid event or mass outbreak. Just trying to figure out how to update the corresponding pickle files and what information is required to accomplish this. I tried to write a python script to read a write to the pickle files but didn’t have any success. 

I saw a couple weeks back a discussion where someone asked for the pickle file to be updated or something like that but I haven’t been able to find anything else.

**Reply by kwsch** [reply]
Again, no relation to python's format.

pkNX is the tool used to dump for outbreaks and raids, which uses the ROM data to brute force the results.
